sdk/src/androidTest/java/com/mapbox/maps/ViewAnnotationTest.kt

```diff
@@ -0,0 +1,683 @@
+          assertTrue(mapView.hasChildView(secondView))
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).x, secondView.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).y, secondView.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
+          latch.countDown()
+        },
+        VIEW_PLACEMENT_DELAY_MS
+      )
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  // checking some use-cases when updating view annotations
+
+  @Test
+  fun updateViewAnnotation() {
+    val latch = CountDownLatch(1)
+    val offsetY = 30
+    mainHandler.post {
+      val view = viewAnnotationManager.addViewAnnotation(
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(CAMERA_CENTER)
+        }
+      )
+      viewAnnotationManager.updateViewAnnotation(
+        view = view,
+        options = viewAnnotationOptions {
+          geometry(SHIFTED_CENTER)
+          offsetY(offsetY)
+        }
+      )
+      mainHandler.postDelayed(
+        {
+          assertTrue(mapView.hasChildView(view))
+          assertEquals(mapboxMap.pixelForCoordinate(SHIFTED_CENTER).x - view.width / 2.0, view.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertEquals(mapboxMap.pixelForCoordinate(SHIFTED_CENTER).y - view.height / 2.0 - offsetY, view.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
+          latch.countDown()
+        },
+        VIEW_PLACEMENT_DELAY_MS
+      )
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
+  }
+
+  @Test
+  fun updateViewAnnotationDimensions() {
+    val latch = CountDownLatch(1)
+    val updatedWidthPx = 120
+    val updatedHeightPx = 150
+    mainHandler.post {
+      val view = viewAnnotationManager.addViewAnnotation(
+        resId = R.layout.view_annotation,
+        options = viewAnnotationOptions {
+          geometry(CAMERA_CENTER)
+        }
+      )
+      viewAnnotationManager.updateViewAnnotation(
+        view = view,
+        options = viewAnnotationOptions {
+          width(updatedWidthPx)
+          height(updatedHeightPx)
+        }
+      )
+      mainHandler.postDelayed(
+        {
+          assertTrue(mapView.hasChildView(view))
+          assertEquals(updatedWidthPx, view.width)
+          assertEquals(updatedHeightPx, view.height)
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).x - view.width / 2.0, view.translationX.toDouble(), ADMISSIBLE_ERROR_PX)
+          assertEquals(mapboxMap.pixelForCoordinate(CAMERA_CENTER).y - view.height / 2.0, view.translationY.toDouble(), ADMISSIBLE_ERROR_PX)
+          latch.countDown()
+        },
+        VIEW_PLACEMENT_DELAY_MS
+      )
+    }
+    if (!latch.await(CONFIGURE_TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+      throw TimeoutException()
+    }
```

Comment:
```suggestion
latch.throwIfNotAwait()
```

```
    fun CountDownLatch.throwIfNotAwait(timeoutMs = CONFIGURE_TIMEOUT_MS){
           if (!this.await(timeoutMs, TimeUnit.MILLISECONDS)) {
               throw TimeoutException()
           }
    }
```
and add to some utils class/file

Reply:
done